lib/charms/loki_k8s/v1/loki_push_api.py

```diff
@@ -2313,6 +2396,156 @@ def _containers(self) -> Dict[str, Container]:
         return {cont: self._charm.unit.get_container(cont) for cont in self._logs_scheme.keys()}
 
 
+class LogForwarder(ConsumerBase):
+    """Forward the standard outputs of all workloads operated by a charm to one or multiple Loki endpoints."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        *,
+        relation_name: str = DEFAULT_RELATION_NAME,
+        loki_endpoints_getter: Optional[Callable[[Relation], List[str]]] = None,
```

Comment:
This should be reverted to just a Dict[str,str] in the form of `{unit_name: url}` instead of a Callable. And it should be called `loki_endpoints`.